Source/Core/Core/HW/SI/SI.cpp

```diff
@@ -595,6 +596,8 @@ void ChangeDeviceDeterministic(SIDevices device, int channel)
 
 void UpdateDevices()
 {
+  s_polled_from_vi = true;
```

Comment:
I don't really get how this works. Seems like the only way s_polled_from_vi could be true is if you check it while UpdateDevices() is being ran. Is there anything in this function that's blocking?

Also, from what I can tell, this is meant to be checked from a separate thread -- it should likely be a Common::Flag.

Reply:
It's actually on the same thread, as `NetPlay_GetInput` is called from SI. Maybe it would make more sense to move this variable into the NetPlay namespace?